tests/func/test_api.py

```diff
@@ -5,93 +5,42 @@
 from dvc import api
 from dvc.api import UrlNotDvcRepoError
 from dvc.exceptions import FileMissingError
-from dvc.main import main
 from dvc.path_info import URLInfo
 from dvc.utils.fs import remove
-from tests.remotes import (
-    GCP,
-    HDFS,
-    OSS,
-    S3,
-    SSH,
-    TEST_REMOTE,
-    Azure,
-    GDrive,
-    Local,
-)
-
-remote_params = [S3, GCP, Azure, GDrive, OSS, SSH, HDFS]
-all_remote_params = [Local] + remote_params
-
-
-@pytest.fixture
-def remote_url(request):
-    if not request.param.should_test():
-        raise pytest.skip()
-    return request.param.get_url()
-
-
-def run_dvc(*argv):
-    assert main(argv) == 0
-
-
-def ensure_dir(dvc, url):
-    if url.startswith("gdrive://"):
-        GDrive.create_dir(dvc, url)
-        run_dvc(
-            "remote",
-            "modify",
-            TEST_REMOTE,
-            "gdrive_service_account_email",
-            "test",
-        )
-        run_dvc(
-            "remote",
-            "modify",
-            TEST_REMOTE,
-            "gdrive_service_account_p12_file_path",
-            "test.p12",
-        )
-        run_dvc(
-            "remote",
-            "modify",
-            TEST_REMOTE,
-            "gdrive_use_service_account",
-            "True",
-        )
-
-
-def ensure_dir_scm(dvc, url):
-    if url.startswith("gdrive://"):
-        GDrive.create_dir(dvc, url)
-        with dvc.config.edit() as conf:
-            conf["remote"][TEST_REMOTE].update(
-                gdrive_service_account_email="test",
-                gdrive_service_account_p12_file_path="test.p12",
-                gdrive_use_service_account=True,
-            )
-        dvc.scm.add(dvc.config.files["repo"])
-        dvc.scm.commit(f"modify '{TEST_REMOTE}' remote")
-
```

Comment:
Finally this moved into the fixture itself. But the fixture itself could be  and will be improved further later.